docs/modules/ROOT/pages/references/principles.adoc

```diff
@@ -0,0 +1,18 @@
+= Design principles
+
+* An instance of Keycloak is responsible for authoritatively store data on:
+** Organizations
+** Teams
+** Users
+
+* We make use of https://kyverno.io/[Kyverno] to enforce policies that go beyond the capabilities of RBAC.
+  In the case Kyverno is not suitable, we will create a custom admission/mutating admission webhook for exactly that use case.
+
+* In cases where data needs to be copied to other locations, we will only do so in one direction.
+  
+* Data owned by Keycloak will be cached on each Zone.
+  This for once is required for Kyverno to work.
+  It also makes the cluster more resilient against connectivity issues between Keycloak and the Zone.
```

Comment:
I would argue that this isn't a principle, but rather in general our current system idea/proposal. I'd like to come up with a different proposal and a diagram supporting this text.